README.md

```diff
@@ -582,6 +582,32 @@ func main() {
         // all is good, use result
     }
 }
+```
+Here's an example:
+```go
+type number struct {
+	n int
+}
+
+// define a sqrt function on "number" type that returns
+// a float64 and error value based on the input
+func (N number) sqrt() (val float64, err error) {
+	if N.n < 0 {
+		err = errors.New("Negative value!")
```

Comment:
```suggestion
		err = errors.New("negative value")
```
Error strings should not be capitalized or end with punctuation. See https://github.com/golang/go/wiki/CodeReviewComments#error-strings